### DISCUSSION TITLE
Shell scripting

### DISCUSSION THREAD

**Discussion by natew** [jart/cosmopolitan#113] [Ideas]
Seeing https://github.com/jart/cosmopolitan/issues/104 and https://github.com/jart/cosmopolitan/issues/61 and dealing with bash all day today made me realize how much I'd appreciate an honest try at a better scripting language to replace bash, and how well cosmopolitan works for that.

In a similar vein to how redbean makes simple servers great, I could totally see another *bean that sets up a new and more intuitive shell script env.

Wren has great syntax, but ideally you'd have a lot more shell-y things built in first-class. mruby perhaps has a few libraries that if bundled right would help. [Oil](http://www.oilshell.org/release/latest/doc/idioms.html) looks interesting and actually targeted to the right use case.

**Comment by jart** [comment]
We've got the Debian Git Almquist Shell checked-in to the examples folder of the codebase, which we call the [UNBOURNE SHELL](https://raw.githubusercontent.com/jart/cosmopolitan/master/examples/unbourne.c). 

```
git clone https://github.com/jart/cosmopolitan && cd cosmopolitan
make -j8 MODE=tiny o/tiny/examples/unbourne.com
o/tiny/examples/unbourne.com
$ echo $(($$ + $$))
27586
$ x=hello
$ echo ${x##*l}
o
```

The issue is it works great on every operating system that isn't Windows. The Almquist Shell is also a subset of bash. People use it because it's faster and meets the minimum requirements for POSIX. So it's not necessarily a more intuitive environment. It's just an example of how Cosmopolitan can absolutely help make your vision possible. Provided that you're ok with programs only running on five operating systems instead of six.
